contracts/src/libraries/EigenDACertVerificationV2Lib.sol

```diff
@@ -1,78 +1,306 @@
+            securityThresholds
+        );
+        if (err != StatusCode.SUCCESS) {
+            return (err, errParams);
+        }
+
+        // Verify blob quorums are a subset of confirmed quorums
+        uint256 blobQuorumsBitmap;
+        (err, errParams, blobQuorumsBitmap) =
+            checkBlobQuorumsSubset(blobInclusionInfo.blobCertificate.blobHeader.quorumNumbers, confirmedQuorumsBitmap);
+        if (err != StatusCode.SUCCESS) {
+            return (err, errParams);
+        }
+
+        // Verify required quorums are a subset of blob quorums
+        return checkRequiredQuorumsSubset(requiredQuorumNumbers, blobQuorumsBitmap);
+    }
+
+    /**
+     * @notice Checks blob inclusion in the batch using Merkle proof
+     * @param batchHeader The batch header
+     * @param blobInclusionInfo The blob inclusion info
+     * @return err Error code (SUCCESS if verification succeeded)
+     * @return errParams Additional error parameters
+     */
+    function checkBlobInclusion(BatchHeaderV2 memory batchHeader, BlobInclusionInfo memory blobInclusionInfo)
+        internal
+        pure
+        returns (StatusCode err, bytes memory errParams)
+    {
+        bytes32 blobCertHash = EigenDAHasher.hashBlobCertificate(blobInclusionInfo.blobCertificate);
+        bytes32 encodedBlobHash = keccak256(abi.encodePacked(blobCertHash));
+        bytes32 rootHash = batchHeader.batchRoot;
+
+        bool isValid = Merkle.verifyInclusionKeccak(
+            blobInclusionInfo.inclusionProof, rootHash, encodedBlobHash, blobInclusionInfo.blobIndex
+        );
+
+        if (isValid) {
+            return (StatusCode.SUCCESS, "");
+        } else {
+            return
+                (StatusCode.INVALID_INCLUSION_PROOF, abi.encode(blobInclusionInfo.blobIndex, encodedBlobHash, rootHash));
+        }
+    }
+
+    /**
+     * @notice Checks the security parameters for a blob cert
+     * @param blobParams The blob params to verify
+     * @param securityThresholds The security thresholds to verify against
+     * @return err Error code (SUCCESS if verification succeeded)
+     * @return errParams Additional error parameters
+     */
+    function checkSecurityParams(VersionedBlobParams memory blobParams, SecurityThresholds memory securityThresholds)
+        internal
+        pure
+        returns (StatusCode err, bytes memory errParams)
+    {
+        if (securityThresholds.confirmationThreshold <= securityThresholds.adversaryThreshold) {
+            return (
+                StatusCode.INVALID_THRESHOLD_PERCENTAGES,
+                abi.encode(securityThresholds.confirmationThreshold, securityThresholds.adversaryThreshold)
+            );
+        }
+        uint256 gamma = securityThresholds.confirmationThreshold - securityThresholds.adversaryThreshold;
+        uint256 n = (10000 - ((1_000_000 / gamma) / uint256(blobParams.codingRate))) * uint256(blobParams.numChunks);
+        uint256 minRequired = blobParams.maxNumOperators * 10000;
+
```

Comment:
a bit late but is there any chance we can comment what this functionality does?

Reply:
I would like full documentation on everything in general, and I want to also add a doc to specify exactly what verifications are being run and how, so that we can add/remove these as needed. But IMO it belongs in another PR.